### PR TITLE
Date fixes

### PR DESCRIPTION
**Pull request by matthewdias**
The `last_read_at` field is supposed to be a query param per https://developer.github.com/v3/activity/notifications/#mark-as-read

This also fixes the `FormattedTimeAdapter` to do proper timezone conversion.

### CODE REVIEW

File: library/src/main/java/com/meisolsson/githubsdk/core/DateQuery.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.meisolsson.githubsdk.core;
+
+import org.joda.time.format.ISODateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.Date;
+
+public class DateQuery {
```
Comment (matthewdias) [inline library/src/main/java/com/meisolsson/githubsdk/core/DateQuery.java:8]: I'm not sure if having this proxy class is okay, or if we would rather the consumer just pass in a date string. With this implementation you have to pass something like `new DateQuery(new java.util.Date())`

**Comment by Meisolsson** [comment]
Not sure that it should be a query parameter. The docs state that 

> For POST, PATCH, PUT, and DELETE requests, parameters not included in the URL should be encoded as JSON with a Content-Type of 'application/json':

[Source](https://developer.github.com/v3/#parameters)

So by that the current implementation is correct. If the time conversion is a problem just open a new PR for that.

**Comment by matthewdias** [comment]
I couldn't get the request to work at all when sending it in body, but it worked properly as a URL param. Was investigating this because it's been broken in Octodroid. When sending a body, notifications do not get dismissed. So maybe this is GitHubs fault if the docs say that, but this change will make the requests work where they did not before.

**Comment by Meisolsson** [comment]
Just testad it from a simple HTTP client and using the body works. Your problem might be down to what time you send.

**Comment by matthewdias** [comment]
Hmm I tested it with different times and none of them worked in body. They return a 205 as expected but have no effect on unread notifications. Just to be sure, it's supposed to mark read any notifs that are older than the time you send, correct?

**Comment by Meisolsson** [comment]
Alright, last thing I can think of is that the time is incorrect relative to UTC. There is some talk about timezones here https://developer.github.com/v3/#timezones

**Comment by matthewdias** [comment]
I did test timezone validity and that's why I ended up fixing the date conversion. But it didn't seem to be the issue for me. If you aren't able to reproduce I can redo my testing when I get back from my trip and provide detailed results.

Edit: it's possible something wonky is going on with the automatic time one detection so I will give that a look as well.

**Comment by Meisolsson** [comment]
Alright, I've just tested it on the latest version and it's working. Used as following.

```
service.markAllNotificationsRead(NotificationReadRequest.builder().build())
```

**Pull request closed without merging**